### PR TITLE
3567 flask feeds controller

### PR DESCRIPTION
**Pull request by tino097**
Fixes #3586 
clean PR for feeds controller

### CODE REVIEW

File: ckan/tests/controllers/test_feed.py [hunk 1]
```diff
@@ -1,147 +1,95 @@
...
-        }
-        app = self._get_test_app()
-        res = app.get(offset, params=params)
-
-        assert '<title>{0}</title>'.format(dataset1['title']) in res.body
-
-        assert '<title>{0}</title>'.format(dataset2['title']) not in res.body
-
-
-class TestFeedInterface(helpers.FunctionalTestBase):
```
Comment (amercader) [inline ckan/tests/controllers/test_feed.py]: We seem to have lost this test class? It's important as it tests a plugin interface, which we want to make sure keeps working as until now

Comment (tino097) [reply ckan/tests/controllers/test_feed.py]: i remove those tests as i had problems to find/rewrite replacement for [webhelpers GeoAtom1Feed](https://bitbucket.org/bbangert/webhelpers/src/9ad434bec9a16c06c1cfeed38cde02f00a95685d/webhelpers/feedgenerator.py?at=trunk&fileviewer=file-view-default#feedgenerator.py-550).

Comment (amercader) [reply ckan/tests/controllers/test_feed.py]: @tino097 just change [this](https://github.com/tino097/ckan/blob/f1f36f8a67b1e247005bb7cb9911b42ff8227af8/ckan/tests/controllers/test_feed.py#L140) to use whatever constructor class you are using in the feeds blueprint now.

**Comment by tino097** [comment]
@amercader can you please check the error within the test and give some hints how to fix it ?

**Comment by amercader** [comment]
@tino097 indentation :)

The first test doesn't create any datasets, so the response code is not reached:

```diff
diff --git a/ckan/views/feed.py b/ckan/views/feed.py
index a09122e..51cdcd6 100644
--- a/ckan/views/feed.py
+++ b/ckan/views/feed.py
@@ -124,10 +124,9 @@ def output_feed(results, feed_title, feed_description, feed_link, feed_url,
                 unicode(len(json.dumps(pkg))), u'application/json'),
             **additional_fields)
 
-        # response.content_type = feed.mime_type
-        resp = make_response(feed.writeString(u'utf-8'), 200)
-        resp.headers['Content-Type'] = u'text/xml'
-        return resp
+    resp = make_response(feed.writeString(u'utf-8'), 200)
+    resp.headers['Content-Type'] = u'text/xml'
+    return resp

```

Content type should be `application/atom+xml`

**Comment by tino097** [comment]
@amercader thank you very much

**Comment by amercader** [comment]
Thanks @tino097!

**Comment by amercader** [comment]
@martinburchell as you were the original contributors of the interface to extend the Atom feeds it would be great if you could test your extension against master to make sure that this refactor maintains compatibility

**Comment by martinburchell** [comment]
@amercader OK, will aim to look in the next week

**Comment by martinburchell** [comment]
@amercader . I've run the tests for https://github.com/aptivate/ckanext-mapactiongeorss against master and they still pass. I've also checked that the custom fields appear correctly in the feed. So this doesn't appear to have broken anything

**Comment by amercader** [comment]
Fantastic, thanks @martinburchell!

**Pull request merged**